osu.Game/Screens/Multi/Screens/Match/Settings/RoomSettingsOverlay.cs

```diff
@@ -0,0 +1,270 @@
+                                Padding = new MarginPadding { Left = field_padding / 2 },
+                                Children = new[]
+                                {
+                                    new Section("MAX PARTICIPANTS")
+                                    {
+                                        Child = MaxParticipantsField = new SettingsTextBox
+                                        {
+                                            RelativeSizeAxes = Axes.X,
+                                            TabbableContentContainer = this,
+                                            OnCommit = (sender, text) => apply(),
+                                        },
+                                    },
+                                    new Section("PASSWORD (OPTIONAL)")
+                                    {
+                                        Child = new SettingsPasswordTextBox
+                                        {
+                                            RelativeSizeAxes = Axes.X,
+                                            TabbableContentContainer = this,
+                                            OnCommit = (sender, text) => apply(),
+                                        },
+                                    },
+                                },
+                            },
+                        },
+                    },
+                    ApplyButton = new ApplySettingsButton
+                    {
+                        Anchor = Anchor.BottomCentre,
+                        Origin = Anchor.BottomCentre,
+                        Size = new Vector2(230, 35),
+                        Margin = new MarginPadding { Bottom = 20 },
+                        Action = apply,
+                    },
+                },
+            };
+
+            TypePicker.Current.ValueChanged += t => typeLabel.Text = t.Name;
+
+            nameBind.ValueChanged += n => NameField.Text = n;
+            availabilityBind.ValueChanged += a => AvailabilityPicker.Current.Value = a;
+            typeBind.ValueChanged += t => TypePicker.Current.Value = t;
+            maxParticipantsBind.ValueChanged += m => MaxParticipantsField.Text = m?.ToString();
+
+            nameBind.BindTo(room.Name);
+            availabilityBind.BindTo(room.Availability);
+            typeBind.BindTo(room.Type);
+            maxParticipantsBind.BindTo(room.MaxParticipants);
+        }
+
+        [BackgroundDependencyLoader]
+        private void load(OsuColour colours)
+        {
+            typeLabel.Colour = colours.Yellow;
+        }
+
+        protected override void PopIn()
+        {
+            // reapply the rooms values if the overlay was completely closed
+            if (content.Y == -1)
+            {
+                nameBind.TriggerChange();
+                availabilityBind.TriggerChange();
+                typeBind.TriggerChange();
+                maxParticipantsBind.TriggerChange();
```

Comment:
This is so that when the user closes the overlay completely with changes in the fields, they are changed back to to `Room`'s values, but not sure if this logic is wanted.